akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -97,7 +97,7 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
         pull(frameIn)
         pull(substreamIn)
 
-        bufferedFrameOut.push(SettingsFrame(Nil)) // server side connection preface
+        bufferedFrameOut.buffer(SettingsFrame(Nil)) // server side connection preface // FIXME don't do this, since we want to GOAWAY right away 
```

Comment:
So removing that push generally speaking is needed so the first frame MAY be GO_AWAY if we want to.
Technically there's no obligation for it to be the first one I guess (the go-away), though the h2spec wants this so in many tests, so would be good to be able to control this fully anyway. I'll read more about Settings semantics too - I could continue on those after this perhaps.